apps/condo/domains/user/schema/User.test.js

```diff
@@ -606,6 +606,113 @@ describe('User fields', () => {
+            expect(userAttrsByUserWithPermissions.rightsSet.connect.id).toEqual(canManageUserRightsSetField.id)
+        })
+
+        test('Support cannot add to user some rightsSet', async () => {
+            const [user] = await createTestUser(admin)
+            const supportUser = await makeClientWithSupportUser()
+            const [canManageUserHasMarketingConsentField] = await createTestUserRightsSet(admin, {
+                canManageUserHasMarketingConsentField: true,
+            })
+            await expectToThrowAccessDeniedErrorToObj(async () => {
+                await updateTestUser(supportUser, user.id, { rightsSet: { connect: { id: canManageUserHasMarketingConsentField.id } } })
+            })
+        })
+    })
+
+    describe('email', () => {
+        let admin
+        let support
+        let UserWithEmail
+        beforeAll(async () => {
+            admin = await makeLoggedInAdminClient()
+            support = await makeClientWithSupportUser()
+            UserWithEmail = generateGQLTestUtils(generateGqlQueries('User', '{ id email deletedAt }'))
+        })
+
+        test('cannot be read by support', async () => {
+            const [user] = await createTestUser(admin)
+            await expectToThrowAccessDeniedError(async () => {
+                await UserWithEmail.getOne(support, { id: user.id })
+            }, ['objs', 0, 'email'])
+        })
+
+        test('can read by user with permissions', async () => {
+            const [user, userAttrs] = await createTestUser(admin)
+            const support = await makeClientWithSupportUser()
+            const [canReadUserEmailField] = await createTestUserRightsSet(admin, {
+                canReadUserEmailField: true,
+            })
+
+            await updateTestUser(admin, support.user.id, { rightsSet: { connect: { id: canReadUserEmailField.id } } })
+
+            const readUser = await UserWithEmail.getOne(support, { id: user.id } )
+            expect(readUser.email).toEqual(userAttrs.email)
+        })
+    })
+
+    describe('phone', () => {
+        let admin
+        let support
+        let UserWithPhone
+        beforeAll(async () => {
+            admin = await makeLoggedInAdminClient()
+            support = await makeClientWithSupportUser()
+            UserWithPhone = generateGQLTestUtils(generateGqlQueries('User', '{ id phone deletedAt }'))
+        })
+
+        test('cannot be read by support', async () => {
+            const [user] = await createTestUser(admin)
+            await expectToThrowAccessDeniedError(async () => {
+                await UserWithPhone.getOne(support, { id: user.id })
+            }, ['objs', 0, 'phone'])
+        })
+
+        test('can read by user with permissions', async () => {
+            const [user, userAttrs] = await createTestUser(admin)
+            const support = await makeClientWithSupportUser()
```

Comment:
again. not a clean test. maybe I can do it because I support and not because I have rightsSet